Fix pointers on ARM

mnussbaum opened the pull request:
Hi there, I noticed this tool doesn't compile correctly on a raspberry pi without the following fix. Thanks for taking a look!

mnussbaum committed:
Fix pointers on ARM

jusw85 reviewed (changes requested):
I don't have an ARM environment to test. 

Please change to the following in `decompress` and `compress` and let me know if it works

```
            block.as_ptr() as *const c_char,
            obuffer.as_mut_ptr() as *mut c_char,
            block.len() as c_int,
            decompressed_size as c_int,
```

```
            ibuffer.as_ptr() as *const c_char,
            obuffer[(magic_number_len + 4)..].as_mut_ptr() as *mut c_char,
            uncompressed_size as c_int,
            compress_bound as c_int,
```

Thanks!

jusw85 committed:
Explicitly specify datatypes for C FFI

c_char and c_int might not be defined as i8 and i32 on some platforms.
Use c_char and c_int explicitly instead of trying to cast to i8 and i32.

mnussbaum commented:
Tested your changes out on a raspberry pi 3b+ and it works well! Thanks for the edits
